spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java

```diff
@@ -120,6 +123,7 @@ public EurekaInstanceConfigBean eurekaInstanceConfigBean(InetUtils inetUtils) {
 		EurekaInstanceConfigBean instance = new EurekaInstanceConfigBean(inetUtils);
 		instance.setNonSecurePort(this.nonSecurePort);
 		instance.setInstanceId(getDefaultInstanceId(this.env));
+		instance.setPreferIpAddress(preferIpAddress);
```

Comment:
Yes, the field will get automatically set by spring **after** `eurekaInstanceConfigBean()` method returns. However, it's too late, `eurekaInstanceConfigBean()` will use the uninitialized `preferIpAddress` to set `statusPageUrlPath`.